drivers/gpu/drm/bridge/ti/fpd_dp_ser_drv.c

```diff
@@ -2134,14 +2187,12 @@ bool fpd_dp_ser_init(void)
 		return false;
 	}
 
-	WRITE_ONCE(deser_reset, 0);
-
 	fpd_dp_ser_set_up_mcu(fpd_dp_priv->priv_dp_client[0]);
 
-	fpd_dp_ser_set_ready(true);
 	fpd_dp_ser_unlock_global();
 
-	queue_work(fpd_dp_priv->motor_setup_wq, &fpd_dp_priv->motor_setup_work);
+	if (wait_display_startup_done(fpd_dp_priv->priv_dp_client[2]))
```

Comment:
Function fpd_dp_ser_init is expected to return false when it fails to initialize serdes.